queue.go

```diff
@@ -0,0 +1,36 @@
+package queue
+
+import (
+	"fmt"
+)
+
+type Song string
+
+type Queue struct {
+	 songList map[Song]int
+}
+
+func NewQueue() *Queue {

```

Comment:
Add comments [per the go style guid](https://golang.org/doc/effective_go.html#commentary)